From 924ceaed70a63e5eaaf39b78dc756e4a7ac3c250 Mon Sep 17 00:00:00 2001
From: Weilong Chen <chenweilong@huawei.com>
Date: Fri, 3 Dec 2021 18:15:45 +0800
Subject: [PATCH] ACPI / APEI: Notify all ras err to driver
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: 924ceaed70a63e5eaaf39b78dc756e4a7ac3c250


ascend inclusion
category: feature
bugzilla: https://gitee.com/openeuler/kernel/issues/I4CMAR
CVE: NA

-------------------------------------------------

Customization deliver all types error to driver. As the driver
need to process the errors in process context.

Signed-off-by: Weilong Chen <chenweilong@huawei.com>
Reviewed-by: Xie XiuQi <xiexiuqi@huawei.com>
Signed-off-by: Yang Yingliang <yangyingliang@huawei.com>
Reviewed-by: Weilong Chen <chenweilong@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 drivers/acpi/apei/ghes.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/drivers/acpi/apei/ghes.c b/drivers/acpi/apei/ghes.c
index 744769f7bddb..fc816c902394 100644
--- a/drivers/acpi/apei/ghes.c
+++ b/drivers/acpi/apei/ghes.c
@@ -665,11 +665,13 @@ static bool ghes_do_proc(struct ghes *ghes,
 		} else {
 			void *err = acpi_hest_get_payload(gdata);
 
-			ghes_defer_non_standard_event(gdata, sev);
 			log_non_standard_event(sec_type, fru_id, fru_text,
 					       sec_sev, err,
 					       gdata->error_data_length);
 		}
+
+		/* Customization deliver all types error to driver. */
+		ghes_defer_non_standard_event(gdata, sev);
 	}
 
 	return queued;
-- 
2.26.2

